pkg/module/drift/terraform.go

```diff
@@ -2,19 +2,85 @@ package drift
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"os"
 	"reflect"
 	"strings"
+	"time"
 
+	"github.com/cloudquery/cloudquery/pkg/module/drift/terraform"
 	"github.com/cloudquery/cq-provider-sdk/provider/schema"
 	"github.com/doug-martin/goqu/v9"
 	"github.com/doug-martin/goqu/v9/exp"
 	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/olekukonko/tablewriter"
 )
 
-func (d *Drift) driftTerraform(ctx context.Context, conn *pgxpool.Conn, cloudName string, cloudTable *traversedTable, resName string, resources map[string]*ResourceConfig, iacData *IACConfig) (*Result, error) {
+type TFStates []*terraform.Data
+
+// FindType returns all instances of the given type under a given mode
+func (t TFStates) FindType(tfType, tfMode string) TFInstances {
+	var ret []terraform.Instance
+	for _, d := range t {
+		for idx, r := range d.State.Resources {
+			if tfMode != "" && r.Mode != tfMode {
+				continue
+			}
+			if r.Type != tfType {
+				continue
+			}
+			ret = append(ret, d.State.Resources[idx].Instances...)
+		}
+	}
+	return ret
+}
+
+type TFInstances []terraform.Instance
+
+const tfIDAttribute = "id"
+
+// Attributes returns a map of resource ID vs. attributes
+func (r TFInstances) AsResourceList(attributeNames []string, attributeTypes []schema.ValueType) ResourceList {
+	ret := make([]*Resource, len(r))
+	for i := range r {
+		var attributes map[string]interface{}
+		if err := json.Unmarshal(r[i].AttributesRaw, &attributes); err != nil {
+			panic(err)
+		}
+
+		res := &Resource{
+			ID: attributes[tfIDAttribute].(string),
+		}
+		res.Attributes = make([]interface{}, len(attributeNames))
+		for i := range attributeNames {
+			if val, ok := attributes[attributeNames[i]]; ok {
+				res.Attributes[i] = parseTerraformAttribute(val, attributeTypes[i])
+			}
+		}
+
+		ret[i] = res
+	}
+	return ret
+}
+
+func parseTerraformAttribute(val interface{}, t schema.ValueType) interface{} {
+	switch t {
+	case schema.TypeTimestamp:
+		ts, err := time.Parse(time.RFC3339, val.(string))
+		if err != nil {
+			ts, err = time.Parse("2006-01-02 15:04:05 -0700 MST", val.(string))
+		}
+		if err != nil {
+			return val // will probably error/detect deep drift
+		}
+		return fmt.Sprintf("%d", ts.Unix())
+	default:
+		return val
+	}
+}
+
+func (d *Drift) driftTerraform(ctx context.Context, conn *pgxpool.Conn, cloudName string, cloudTable *traversedTable, resName string, resources map[string]*ResourceConfig, iacData *IACConfig, states TFStates) (*Result, error) {
```

Comment:
I think this function has a very big signature, I think if you break it down, it doesn't use the Drift object much, but this would require a bigger refactor/change up, which is out of this PR's scope imo.

Reply:
started on that, might do more later in another PR